### PR TITLE
Add deno otel logging support

### PR DESCRIPTION
**Pull request by MikeShi42**
Creates a deno custom logger compatible with the std/log module that emits via the new otel js logging api.

### CODE REVIEW

**Comment by wrn14897** [comment]
Cool stuff. I did a quick test and found a few bugs:

1. the service name is not set properly when `detectResources` = false
2. the structured log is not parsed. It looks like
```
INFO {"message":"This is a warning"}
```
Although we can make the ingestor smarter, it would be nice to strip the level from the message

File: packages/deno/mod.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  BufferConfig,
+  LoggerProvider,
+} from 'npm:@opentelemetry/sdk-logs@0.43.0';
+
+import type { Logger } from 'npm:@opentelemetry/api-logs@0.43.0';
+import type { Attributes } from 'npm:@opentelemetry/api@1.6.0';
+
+// https://github.com/open-telemetry/opentelemetry-specification/blob/fc8289b8879f3a37e1eba5b4e445c94e74b20359/specification/logs/data-model.md#displaying-severity
+const OTEL_SEVERITY_NAME_MAP = {
+  0: '', // XXX: This is not to spec
```
Comment (MikeShi42) [inline packages/deno/mod.ts:33]: @wrn14897 I forgot to ask you this - but what's our current behavior for a undefined level? Should we keep it as empty string or something else? It's undefined in the otel spec and I'm not sure if we do something about this elsewhere.

We do need _something_ here though I think to keep the lookup table complete.

Comment (wrn14897) [reply packages/deno/mod.ts:33]: interesting, I don't recall anything regarding undefined level but I'm going to make sure we address that

Comment (wrn14897) [reply packages/deno/mod.ts:33]: I think the default is 'info' if severity text is undefined or empty

**Review by wrn14897 — APPROVED** [review]
🚢 🚢 🚢

**Pull request merged**